examples/vision/siamese.py

```diff
@@ -0,0 +1,232 @@
+
+"""
+# Siamese Network with Contrastive Loss
+
+Author: [Mehdi](https://github.com/s-mrb)<br>
+Date created: 2021/05/02<br>
+Last modified: 2020/05/03<br>
+Description: Similarity Learning Using Siamese Network with Contrastive Loss
+"""
+
+"""
+## Introduction
+
+The [Siamese Network](https://papers.nips.cc/paper/1993/file/288cc0ff022877bd3df94bc9360b9c5d-Paper.pdf)
+introduced by Bromley and LeCun to solve signature verification as an image
+matching problem is now widely used for finding the level of similarity between
+two images. This algorithm evolved to have many different forms, although
+the one we are going to use is Duplet Network. A duplet network involves
+two sister networks for finding embeddings/encodings and then a distance
+heuristic to find out how much these embeddings differ from each other.
```

Comment:
Suggestion:

- Minor refactoring to improve readability
- Keeping 80 characters per line
 
```suggestion
The [Siamese network](https://papers.nips.cc/paper/1993/file/288cc0ff022877bd3df94bc9360b9c5d-Paper.pdf)
- introduced by Bromley and LeCun in 1993 to solve signature verification as an
image matching problem - is now widely used for finding the level of similarity
between two images. This algorithm evolved to have many different forms,
although the one we are going to use in this example is called the Duplet
network. The Duplet network involves: 1) two sister networks for finding
embeddings/encodings; and then 2) a distance heuristic to find out how much these
embeddings differ from each other.
```

Reply:
It may be worth borrowing some ideas to help the users understand what a Siamese network is from the following sources: 1) https://keras.io/examples/vision/siamese_network/; and 2) https://keras.io/examples/vision/siamese_network/